compiler/src/dotty/tools/io/Path.scala

```diff
@@ -141,12 +144,12 @@ class Path private[io] (val jfile: JFile) {
       // the only solution <-- a comment which could have used elaboration
       if (segments.nonEmpty && segments.last == "..")
         (path / "..").toDirectory
-      else jfile.getParent match {
+      else jpath.getParent match {
         case null =>
           if (isAbsolute) toDirectory // it should be a root. BTW, don't need to worry about relative pathed root
           else Directory(".")         // a dir under pwd
         case x    =>
-          Directory(x)
+          Directory(x.toFile)
```

Comment:
There should probably be a constructor that takes a `JPath`. And I think we should remove the `JFile` constructor